pkg/deploy/lattice/listener_synthesizer.go

```diff
@@ -80,6 +89,45 @@ func (l *listenerSynthesizer) Synthesize(ctx context.Context) error {
 	return nil
 }
 
+func (l *listenerSynthesizer) getLatticeListenerDefaultAction(ctx context.Context, modelListenerProtocol string) (
```

Comment:
Thanks for you comment! I did some changes in the new revision and that did make the code cleaner: 

1. Added the field `DefaultAction` in the `modelListener.ListenerSpec` https://github.com/aws/aws-application-networking-k8s/blob/cd6072c039fed864ec7042ee6e38ab348f1f528d/pkg/model/lattice/listener.go#L13-L32

2. Completely skip to create stackRule for TLS_PASSTHROUGH gwListener. i.e,  completely don't call  `atticeServiceModelBuildTask.buildRules()` for  TLS_PASSTHROUGH gwListener, therefore removing the "long comments" in the `buildRules()`  https://github.com/aws/aws-application-networking-k8s/blob/cd6072c039fed864ec7042ee6e38ab348f1f528d/pkg/gateway/model_build_lattice_service.go#L89-L92

3. Added the ` (t *latticeServiceModelBuildTask) getListenerDefaultAction` in the `model_build_listener` https://github.com/aws/aws-application-networking-k8s/blob/cd6072c039fed864ec7042ee6e38ab348f1f528d/pkg/gateway/model_build_listener.go#L128-L151 (Added unit test to cover build TLS_PASSTHROUGH listener)

4. Changed the logic of `(l *listenerSynthesizer) getLatticeListenerDefaultAction(` https://github.com/aws/aws-application-networking-k8s/blob/cd6072c039fed864ec7042ee6e38ab348f1f528d/pkg/deploy/lattice/listener_synthesizer.go#L92-L127 , removed the seemly confusing part `l.tgManager.ResolveRuleTgIds(ctx, stackRules[0], l.stack)`, change to ` l.tgManager.ResolveRuleTgIds(ctx, stackListener.Spec.DefaultAction.Forward, l.stack);()` instead.

5. Still keep the defaultAction argument for the `listenerManager.Upsert()` , since the in the listenerSynthesizer it need to convert the `modelListenerDefaultAction` into the `vpcLatticeListenerDefaultAction`, and pass the `vpcLatticeListenerDefaultAction` to `listenerManager.Upsert()`  to determine (`needToUpdateDefaultAction()`) whether the listener need to be updated.



Ran the full test suite for the new revision(skip the RAM test):
```
• [257.448 seconds]
------------------------------
[SynchronizedAfterSuite]
/Volumes/workspace/aws-application-networking-k8s/test/suites/integration/suite_test.go:72
{“level”:“info”,“ts”:“2024-06-05T23:08:51.716-0700",“caller”:“test/framework.go:264",“msg”:“Deleting objects: *v1.Gateway/test-gateway, *v1.Pod/grpc-runner”}
{“level”:“info”,“ts”:“2024-06-05T23:08:51.758-0700",“caller”:“test/framework.go:282",“msg”:“Waiting for NotFound, objects: *v1.Gateway/test-gateway, *v1.Pod/grpc-runner”}
[SynchronizedAfterSuite] PASSED [31.193 seconds]
------------------------------
Ran 66 of 69 Specs in 2412.270 seconds
SUCCESS! -- 66 Passed | 0 Failed | 0 Pending | 3 Skipped
--- PASS: TestIntegration (2412.27s)
PASS
```